tests/testcases/config-variables/src/lib.rs

```diff
@@ -12,15 +12,21 @@ fn handle_request(req: Request) -> Result<Response> {
         .query()
         .expect("Should have a password query string");
     let query: std::collections::HashMap<String, String> = serde_qs::from_str(query)?;
-    let provided_password = query
+    let expected_password = query
         .get("password")
         .expect("Should have a password query string");
-    let expected_password = config::get("password")?;
+    let actual_password = config::get("password")?;
 
-    ensure!(
-        provided_password == &expected_password,
-        "password must match expected"
-    );
+    if expected_password != &actual_password {
+        let error_msg = format!(
+            "expected password: '{}', actual password: '{}'",
+            expected_password, &actual_password
+        );
+
+        return Ok(http::Response::builder()
+            .status(500)
```

Comment:
should this be a `401` instead of `500`? Does `ensure!` always throw a `500`?

Reply:
the return code does not really matter as long as it is not `200`. I think ensure panics and that triggers a 500 error code

Reply:
That makes sense that a panic is 500. Thanks. I still would think 401 makes more sense here since it is a known potential error. i understand that it does not matter in the sense that the test will check the error code regardless

Reply:
sure, updated it to 401 :)